src/game/game-admin.guard.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
+import { UsersService } from 'src/users/users.service';
+import { UserEntity as User } from '../users/entities/user.entity';
+
+/**
+ * Check if the user performing the request has admin role.
+ */
+@Injectable()
+export class GameAdminGuard implements CanActivate {
```

Comment:
Rename to `RestAdminGuard` and move to `common/guards`. It is something we might use in other (possibly) REST endpoints, right?